cf/src/main/scala/com/timeout/scalacloudformation/Encoding.scala

```diff
@@ -43,30 +50,60 @@ object Encoding {
       throw new Exception(s"Unexpected expression $x")
   }
 
-  implicit val encodeHNil: ObjectEncoder[HNil] =
-    ObjectEncoder.instance[HNil](_ => JsonObject.empty)
+  implicit def encodeEnum[T: Enum]
+    : Encoder[T] = Encoder.instance[T] { t =>
+    Json.fromString(implicitly[Enum[T]].encode(t))
+  }
 
-  implicit def encodeHList[K <: Symbol, H, T <: HList](
-    implicit
-    encodeHead: Lazy[Encoder[H]],
-    encodeTail: ObjectEncoder[T],
-    witness: Witness.Aux[K]
-  ): ObjectEncoder[FieldType[K, H] :: T] = ObjectEncoder.instance[FieldType[K, H] :: T] {
-    case h :: t =>
-      val obj = encodeTail.encodeObject(t).+:(witness.value.name -> encodeHead.value(h))
-      obj.filterKeys(_ != "logicalId")
+  implicit val encodeParamDataType: Encoder[Parameter.DataType] = {
+    import Parameter.DataType._
+    val awsEnum = implicitly[Enum[Parameter.AwsType]]
+
+    Encoder.instance[Parameter.DataType] { dt =>
+      val s = dt match {
```

Comment:
I tried using `scala.reflect`'s `WeakTypedTag#tpe.typeSymbol.name`, but unfortunately it does return fairly weird string values such us "ListLtNumberGt". So this ends up being the most pragmatic way in this case